doc/flakes/design.md

```diff
@@ -2,92 +2,80 @@
+* A flake can contain a *lock file* (`flake.lock`) used when resolving the
+  dependencies in `flake.nix`. It maps mutable flake references
+  (e.g. `github:NixOS/nixpkgs/release-18.09`) to references containing revisions
+  (e.g. `nixpkgs ->
   github:NixOS/nixpkgs/98a2a5b5370c1e2092d09cb38b9dcff6d98a109f`).
 
-* The `nix` command uses the flake registry as its default
-  installation source. For example, `nix build nixpkgs.hello` builds the
-  `hello` package provided by the `nixpkgs` flake listed in the
-  registry. `nix` will automatically download/upload the registry and
-  flakes as needed.
+* The `nix` command uses the flake registry as its default installation source.
+  For example, `nix build nixpkgs.hello` builds the `hello` package provided by
+  the `nixpkgs` flake listed in the registry. `nix` will automatically
+  download/upload the registry and flakes as needed.
 
 * `nix build` without arguments will build the flake in the current
   directory (or some parent).
 
-* The command `nix flake update` generates/updates `flake.lock` from
-  `flake.nix`. This should probably also be done automatically when
-  building from a local flake.
+* `nix flake update` generates `flake.lock` from `flake.nix`, ignoring the old
+  lockfile.
 
-* `nixos-rebuild` will build a configuration from a (locked)
-  flake. Evaluation will be done in pure mode to ensure there are no
-  unaccounted inputs. Thus the NixOS configuration can be reproduced
-  unambiguously from the top-level flake.
+* `nixos-rebuild` will build a configuration from a (locked) flake. Evaluation
+  is done in pure mode to ensure there are no unaccounted inputs. Thus the
+  NixOS configuration can be reproduced unambiguously from the top-level flake.
 
-* Nix code can query flake metadata such as `commitHash` (the Git
-  revision) or `date` (the date of the last commit). This is useful
-  for NixOS to compute the NixOS version string (which will be the
-  revision of the top-level configuration flake, uniquely identifying
-  the configuration).
+* Nix code can query flake metadata such as `commitHash` (the Git revision) or
+  `epoch` (the date of the last commit). This is useful for NixOS to compute
+  the NixOS version string (which will be the revision of the top-level
+  configuration flake, uniquely identifying the configuration).
 
-* Hydra jobset configurations will consist of a single flake
-  reference. Thus we can get rid of jobset inputs; any other needed
-  repositories can be fetched by the top-level flake. The top-level
-  flake can be locked or unlocked; if some dependencies are unlocked,
-  then Nix will fetch the latest revision for each.
+* Hydra jobset configurations will consist of a single flake reference. Thus we
+  can get rid of jobset inputs; any other needed repositories can be fetched by
+  the top-level flake. The top-level flake can be locked or unlocked; if some
+  dependencies are unlocked, then Nix will fetch the latest revision for each.
```

Comment:
The top-level flake currently must be locked. This is because Hydra has no way to commit/track the lockfile entries it computed.